### PR TITLE
feat/HNG 51 create Integration Card

### LINKED ISSUE #51: [FEATURE] Create the Integrations Card component on the Integrations Page- Admin Dashboard
Develop a card component that displays platform details including a heading, logo, description, and a toggle switch. The toggle switch allows admins to activate or deactivate the platform. Integration with backend endpoints will ensure the toggle switch reflects the current platform status. The component should be responsive and match the design specifications.

**Purpose**
Admins should be able to view and manage a platform activation status through a card component on the dashboard. This component will display platform details including a heading, logo, description, and a toggle switch to activate or deactivate the platform. Integration with the appropriate endpoints will ensure real-time status updates.

### Acceptance Criteria
**Requirement**
Develop a reusable component that includes the following props:
- Heading: Displaying the platform name or title.
- Logo: Displaying the platform logo or icon.
- Description: Providing a brief overview of the platform
- Toggle Switch: Allowing admins to activate or deactivate the platform.
- Toggle Switch Functionality: Implement functionality to toggle the platform activation status.
- Integration with API Endpoint: Ensure the toggle switch updates the platform status via the appropriate API endpoint.
- Responsive Design: Make sure the card component is responsive and adapts well to different screen sizes.
- Design Match: Ensure the card component matches the design provided, including layout and styling.

**Task**

- [x]  Develop a reusable card component with heading, logo, description, and toggle switch
- [x] Ensure the component structure and layout match the design mockups.
- [ ]  Integrate the toggle switch with the backend API endpoint.
- [x] Implement logic to fetch and update the platform activation status.
- [x] Ensure the toggle switch state reflects the current platform status.
- [x] Ensure the card component is responsive.
- [x] Write unit tests to verify toggle switch functionality and API integration.

**Note**

- Ensure code quality through proper code structure and comments
- Provide clear documentation on component usage and integration.

Deadline: Friday 10am

Images and Figma Links
https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=118-34845&m=dev
image

### PR DESCRIPTION
**Pull request by Vaden89**
# Description
Created and Test the Integration Cards need on the integration dashboard, I didn't add the API cause it's not available 

**Closes #51**

# Changes proposed

## What were you told to do?
I was to create and Test Integration card component with toggle functionality 
Figma link: https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=118-34845&m=dev

## What did you do?

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

 

- [X] Develop a reusable card component with heading, logo, description, and toggle switch
- [X]  Ensure the component structure and layout match the design mockups.
- [X]  Integrate the toggle switch with the backend API endpoint.
- [X]  Implement logic to fetch and update the platform activation status.
- [X]  Ensure the toggle switch state reflects the current platform status.
- [X]  Ensure the card component is responsive.
- [X]  Write unit tests to verify toggle switch functionality and API integration.

# Screenshots/Videos

https://github.com/user-attachments/assets/01ebc2ff-be8d-4a84-b1cf-72daf867da8a

Screenshot (49)
Screenshot (50)

### CODE REVIEW

File: src/components/common/IntegrationCard/IntegrationCard.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  };
+
+  return (
+    <div
+      data-testid={INTEGRATION_CARD_TEST_ID}
+      className="flex h-[159.29px] w-[341px] flex-col gap-6 rounded-lg border-[1px] border-border px-4 py-6 text-foreground"
+    >
+      <div className="flex w-full items-center justify-between">
+        <Image
+          data-testid={INTEGRATION_CARD_IMAGE_TEST_ID}
```
Comment (skeby) [inline src/components/common/IntegrationCard/IntegrationCard.tsx]: there's no need for data-testid

File: src/components/common/IntegrationCard/IntegrationCard.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      setisActive(false);
+    } else {
+      setisActive(true);
+    }
+  };
+
+  return (
+    <div
+      data-testid={INTEGRATION_CARD_TEST_ID}
+      className="flex h-[159.29px] w-[341px] flex-col gap-6 rounded-lg border-[1px] border-border px-4 py-6 text-foreground"
```
Comment (skeby) [inline src/components/common/IntegrationCard/IntegrationCard.tsx]: This component wouldn't be responsive. Use full width, then you can set the max width as 341px

**Review by skeby — DISMISSED** [review]
Please attach screenshots to show the result of running `pnpm lint`

File: src/components/common/IntegrationCard/IntegrationCard.tsx (hunk 1 above)
Comment (Vaden89) [reply src/components/common/IntegrationCard/IntegrationCard.tsx]: Removed

File: src/components/common/IntegrationCard/IntegrationCard.tsx (hunk 2 above)
Comment (Vaden89) [reply src/components/common/IntegrationCard/IntegrationCard.tsx]: Done

**Review by taiwonaf — APPROVED** [review]

**Pull request merged**